### DISCUSSION TITLE
Skeleton UI support for Zag

### DISCUSSION THREAD

**Discussion by endigo9740** [chakra-ui/zag#1714] [General]
Hey everyone, my name is [Chris](https://github.com/endigo9740) 👋

I'm the creator and core maintainer for a UI library called [Skeleton](https://www.skeleton.dev/). Over the past couple years we've built a good size audience providing our open source Tailwind-based design system + Svelte component library. Then, earlier this year, we announced that we would be expanding support to both React and Svelte v5 as part of our next major release. We're dubbing this [Skeleton Next](https://github.com/skeletonlabs/skeleton/discussions/2681) (aka v3) and it represents a massive step forward for the project. Including a major refactor and overhaul to most of the library. See our (work in progress) docs here:
https://next.skeleton.dev/

Unfortunately, as part of this process we've run into the sorts of challenges you face when trying to build and maintain parity between components in multiple frameworks. Something that will likely only get worse as we introduce support for Vue, Solid, and others in the future. Thankfully we were recently introduced to Zag, and I'm excited to say it seems like the silver bullet we've been searching for. It seems to solve so many of the issues we faced with component APIs, features, and a11y.

We've been quietly testing this behind the scenes, and finally today I had the channel to announce to our community that we'll be bringing Zag support to our upcoming suite of components. I've provide a full write up here:

- https://github.com/skeletonlabs/skeleton/discussions/2784

I wanted to create this post to introduce myself and the Skeleton project - especially to any Zag team members that might see this. We're very excited about what you folks are doing, and likewise we hope we can be useful as well. Putting your Svelte and React tooling to work in our library, and of course reporting any and all bugs and issues we come across. Likewise, we may even be able to contribute upstream over time.

If anyone is interested, you can contact me directly via [my email](mailto:chris@skeleton.dev). Otherwise we have an active [Discord](https://discord.gg/EXqV7W8MtY) server (username: `endigo9740`), and of course either here or within our [GitHub](https://github.com/skeletonlabs/skeleton) community.

Thank you for what you're doing with Zag. This is super exciting stuff, and I can't wait for our users to get their hands on components using it.

Cheers 🍻

**Comment by segunadebayo** [comment]
Hey Chris,

It's a pleasure to meet you. I am excited that your team is looking to adopt Zag.js to support multiple frameworks better. It is a complex problem to solve and requires immense attention to the details.

I've read through the posts, thought process and even the draft PR in Skeleton. It looks very promising!

Looking forward to seeing the fruits of the collaboration. Let's make Zag.js better together.

**Reply by endigo9740** [reply]
Hey Segun,

Likewise, thanks for the effort to create Zag. We've been on the hunt for something like this for a while. There's some great framework-specific solutions, but the moment you bridge the gap to multi-framework it gets so much more difficult.

We're very excited for what Zag brings to the table and I personally can't wait for users to check our updated components soon. Accessibility is a topic near and dear to my heart, so having a solution that can help improve this is just amazing.

I've always felt that open source works best when teams and individuals collaborate to help make the web better. So I'm very excited to see what we can do together. Great to meet you and I look forward to more interactions in the future!

**Comment by segunadebayo** [comment]
Closing — Skeleton v3.0 has launched with full Zag integration. See #2340.
